packages/io-devtools/src/swag/components/penguin.tsx

```diff
@@ -72,33 +72,36 @@ export const PenguinSpinner: React.FC<PenguinSpinnerProps> = ({
   title = "Sending Transactions",
 }) => {
   const [frameIndex, setFrameIndex] = useState(0);
+  const [isActive, setIsActive] = useState(true);
   const intervalRef = useRef<NodeJS.Timeout | null>(null);
-  const mountedRef = useRef(true);
 
   useEffect(() => {
-    const startAnimation = () => {
-      if (intervalRef.current) {
-        clearInterval(intervalRef.current);
-      }
-
-      intervalRef.current = setInterval(() => {
-        if (mountedRef.current) {
-          setFrameIndex((prev) => (prev + 1) % PENGUIN_FRAMES.length);
-        }
-      }, 500);
-    };
+    if (!isActive) {
+      return;
+    }
 
-    startAnimation();
+    intervalRef.current = setInterval(() => {
+      setFrameIndex((prev) => (prev + 1) % PENGUIN_FRAMES.length);
+    }, 500);
 
     return () => {
-      mountedRef.current = false;
       if (intervalRef.current) {
         clearInterval(intervalRef.current);
         intervalRef.current = null;
       }
     };
+  }, [isActive]);
+
+  useEffect(() => {
```

Comment:
this code is running when component unmounts in the react lifecycle. Its state is about to be destroyed anyway so this is redundant. and could lead to warnings when attempting to change the state of a component which is unmounting